tuf/client/updater.py

```diff
@@ -643,6 +643,13 @@ def refresh(self, unsafely_update_root_if_necessary=True):
         logger.info('An expired Root metadata was loaded and must be updated.')
         raise
 
+    # If an exception is raised during the metadata update attempts, we will
+    # attempt to update root metadata once by recursing with a special argument
+    # to avoid further recursion. We use this bool and pull the recursion out
+    # of the except block so as to avoid unprintable nested NoWorkingMirrorError
```

Comment:
Ah.

It's incorrect here:
https://github.com/theupdateframework/tuf/blob/develop/tuf/client/updater.py#L1152-L1155